packages/devtools_app/lib/src/screens/memory/panes/leaks/primitives/analysis_status.dart

```diff
@@ -0,0 +1,122 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+import '../../../../../primitives/auto_dispose_mixin.dart';
+
+enum AnalysisStatus {
+  NotStarted,
+  Ongoing,
+
+  /// The process is complete and we want user to see the result.
+  ShowingResult,
+
+  /// The process is complete and we want user to acknowledge the error.
+  ShowingError,
+}
+
+Duration _showingResultDelay = const Duration(seconds: 5);
+Duration _delayForUiToHandleState = const Duration(milliseconds: 5);
+
+/// Describes status of the ongoing process.
+class AnalysisStatusController {
+  ValueNotifier<AnalysisStatus> status =
+      ValueNotifier<AnalysisStatus>(AnalysisStatus.NotStarted);
+
+  ValueNotifier<String> message = ValueNotifier('');
+
+  void reset() {
+    status.value = AnalysisStatus.NotStarted;
+    message.value = '';
+  }
+}
+
+/// Shows [analysisStarter] if the analysis is not started yet and the status
+/// of the process otherwise.
+/// If process is completed successfully, keeps the status for
+/// [_showingResultDelay] and then shows [analysisStarter].
+/// If the process ended up with error, show the error and two buttons
+/// (Copy ans OK). After user clicks [OK], shows [analysisStarter].
+class AnalysisStatusView extends StatefulWidget {
+  const AnalysisStatusView({
+    Key? key,
+    required this.controller,
+    required this.analysisStarter,
+  }) : super(key: key);
+  final AnalysisStatusController controller;
+  final Widget analysisStarter;
+
+  @override
+  State<AnalysisStatusView> createState() => _AnalysisStatusViewState();
+}
+
+class _AnalysisStatusViewState extends State<AnalysisStatusView>
```

Comment:
we should be able to make this a stateless widget and use `DualValueListenableBuilder` to listen to `controller.status` and `controller.message`. If we need to introduce a small delay, we should do that where the data is being processed. We can process data in batches to do this. You can look at the CPU profiler screen or the performance screen for examples, we show a linear progress indicator during the "processing" state. 

In general pushing state into the controller class is a good pattern to follow, and then just have the UI respond to notifiers / listenables on the controller